runtime/lua/vim/fs.lua

```diff
@@ -555,11 +557,17 @@ function M.normalize(path, opts)
     path = path:gsub('%$([%w_]+)', vim.uv.os_getenv)
   end
 
-  -- Convert path separator to `/`
-  path = path:gsub(os_sep_local, '/')
+  if win then
+    -- Convert path separator to `/`
+    path = path:gsub(os_sep_local, '/')
+  end
 
   -- Check for double slashes at the start of the path because they have special meaning
-  local double_slash = vim.startswith(path, '//') and not vim.startswith(path, '///')
+  local double_slash = false
+  if not opts._fast then
+    double_slash = vim.startswith(path, '//') and not vim.startswith(path, '///')
```

Comment:
is `vim.startswith()` slow? I see it uses `vim.validate`, maybe `path:sub()` could be inlined here?

Reply:
I didn't think it would be, but skipping this line achieved a 50% increase (60ms -> 35ms) after the other opts applied.

The important thing to note is that `vim.loader` really does not need most of the logic in this function. Arguably it could work well enough without any normalization, it is just used to increase cache hits for different input paths that point to the same place.